packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -270,30 +270,19 @@ - (void)addScopes:(NSArray<NSString *> *)scopes
 #endif
 }
 
-/// @return @c nil if GoogleService-Info.plist not found and clientId is not provided.
-- (GIDConfiguration *)configurationWithClientIdArgument:(id)clientIDArg
-                                 serverClientIdArgument:(id)serverClientIDArg
-                                   hostedDomainArgument:(id)hostedDomainArg {
-  NSString *clientID;
-  BOOL hasDynamicClientId = [clientIDArg isKindOfClass:[NSString class]];
-  if (hasDynamicClientId) {
-    clientID = clientIDArg;
-  } else if (self.googleServiceProperties) {
-    clientID = self.googleServiceProperties[kClientIdKey];
-  } else {
-    // We couldn't resolve a clientId, without which we cannot create a GIDConfiguration.
+/// @return @c nil if GoogleService-Info.plist not found and runtimeClientIdentifier is not
+/// provided.
+- (GIDConfiguration *)configurationWithClientIdentifier:(NSString *)runtimeClientIdentifier
+                                 serverClientIdentifier:(NSString *)runtimeServerClientIdentifier
+                                           hostedDomain:(id)hostedDomain {
+  NSString *clientID = runtimeClientIdentifier ?: self.googleServiceProperties[kClientIdKey];
```

Comment:
Don't you need to check if `self.googleServiceProperties` is null before getting a key from it? Wouldn't it error/crash if it's null?

Reply:
Nope! `foo[bar]` is just a shorthand for `[foo objectForKeyedSubscript:bar]`, so standard messaging rules (a message to `nil` returns zero'd memory, `nil` in this case) apply.

Reply:
(Notice that the previous serverClientID code was actually already relying on that, so this code wasn't even consistent about the overkill 🙂)